Bottom: 716fb3ae9fc7f3bc14bf8eac1d3bfed1ac8b7dc1
Top:    397cdd01b6602818a46dfc10346785281ba062e6
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-02-01 16:22:25 -0600

Update ci-archive script to get variables from bitbake

The script was previously making assumptions about locations of various
output files. Now it gets those locations from bitbake itself.


---
diff --git a/docker/buildbot-worker-ntel/buildbot/ci-archive.sh b/docker/buildbot-worker-ntel/buildbot/ci-archive.sh
index db5edc4..52b8963 100755
--- a/docker/buildbot-worker-ntel/buildbot/ci-archive.sh
+++ b/docker/buildbot-worker-ntel/buildbot/ci-archive.sh
@@ -4,21 +4,37 @@ MACHINE="$1";shift
 DESTDIR="$1";shift
 TIMESTAMP="$1";shift
 
-workdir="build/tmp-glibc-${MACHINE}"
+# Gather variables from bitbake
+bitbake_vars=$(mktemp)
+bitbake -e 2>&1 > $bitbake_vars
+eval $(grep '^TOPDIR=' $bitbake_vars)
+eval $(grep '^TMPDIR=' $bitbake_vars)
+eval $(grep '^DEPLOY_DIR=' $bitbake_vars)
+eval $(grep '^BUILDHISTORY_DIR=' $bitbake_vars)
+
+# Generate output filename and directory
 rootdir="${MACHINE}.${TIMESTAMP}"
 filename=${rootdir}.tar.gz
 archive="${DESTDIR}/${filename}"
 
-files_to_archive=$(mktemp --tmpdir="${workdir}" "files_to_archive.XXXXXXXXXX")
-
-echo "deploy/glibc-${MACHINE}" >> $files_to_archive
-echo "buildhistory" >> $files_to_archive
-find "${workdir}" -type d -name "temp" >> $files_to_archive
+# Gather files to archive into a text file to feed to tar
+files_to_archive=$(mktemp)
+echo "${DEPLOY_DIR}" >> $files_to_archive
+echo "${BUILDHISTORY_DIR}" >> $files_to_archive
+find "${TMPDIR}" -type d -name "temp" >> $files_to_archive
 echo "$*" >> $files_to_archive
 
 mkdir -p "${DESTDIR}"
 
-exec tar --verbose --create --auto-compress \
-    --transform "s!^!${rootdir}/!" \
+# Remove leading and trailing slashes from TOPDIR
+TOPDIR=${TOPDIR#/}; TOPDIR=${TOPDIR%/}
+
+# Create the archive, substituting path prefixes of TOPDIR into rootdir
+tar --verbose --create --auto-compress \
+    --transform "s!^${TOPDIR}!${rootdir}!" \
     --file "${archive}" \
     --files-from="${files_to_archive}"
+
+# Clean up
+rm $bitbake_vars
+rm $files_to_archive
